augur/util_support/auspice_config.py

```diff
@@ -0,0 +1,234 @@
+        config[new_name] = modify(config[old_name])
+    elif new_name in config:
+        deprecated(f"[config file] top level keys {new_name!r} and {old_name!r} were both present, " \
+                   "however the latter is a deprecated version of the former and will be ignored.")
+        del config[old_name]
+    else:
+        deprecated(f"[config file] top level key {old_name!r} is deprecated. Renaming to {new_name!r}" +
+                   (" (with modifications)" if modify else ""))
+        config[new_name] = modify(config[old_name]) if modify else config[old_name]
+        del config[old_name]
+
+def _parse_color_options(options: Any) -> list[dict]:
+    # parse v1-style 'color_options' & convert to v2-style 'colorings'
+    if not isinstance(options, dict):
+        raise AugurError(f"[config file] The deprecated field 'color_options' must be a dictionary")
+    colorings = []
+    for key, info in options.items():
+        # note: if both legentTitle & menuItem are present then we use the latter. See https://github.com/nextstrain/auspice/issues/730
+        if "menuItem" in info:
+            deprecated("[config file] coloring '{}': 'menuItem' has been replaced with 'title'. Using 'menuItem' as 'title'.".format(key))
+            info["title"] = info["menuItem"]
+            del info["menuItem"]
+        if "legendTitle" in info:
+            if "title" in info: # this can only have been set via menuItem ^^^
+                deprecated("[config file] coloring '{}': 'legendTitle' has been replaced with 'title' & is unused since 'menuItem' is present.".format(key))
+            else:
+                deprecated("[config file] coloring '{}': 'legendTitle' has been replaced with 'title'. Using 'legendTitle' as 'title'.".format(key))
+                info["title"] = info["legendTitle"]
+            del info["legendTitle"]
+        if "key" in info:
+            del info["key"]
+        if info.get("type") == "discrete":
+            deprecated("[config file] coloring '{}': type 'discrete' is no longer valid. Please use either 'ordinal', 'categorical' or 'boolean'. "
+                "This has been automatically changed to 'categorical'.".format(key))
+            info["type"] = "categorical"
+        colorings.append({"key": key, **info})
+    return colorings
+
+def _rename_deprecated_authors_coloring(colorings: list):
+    # The 'authors' coloring (plural) is now called 'author' (singular)
+    author  = next(iter([{'index': idx, 'coloring': c} for [idx, c] in enumerate(colorings) if c['key']=='author'] or [None]))
+    authors = next(iter([{'index': idx, 'coloring': c} for [idx, c] in enumerate(colorings) if c['key']=='authors'] or [None]))
+    if authors:
+        if author:
+            deprecated("[config file] coloring 'authors' has been renamed 'author', however a " +
+                       "coloring for 'author' already exists. 'authors' will be ignored")
```

Comment:
```suggestion
            deprecated("[config file] ignoring deprecated coloring 'authors' because a " +
                       "coloring for 'author' already exists.")
```

Reply:
Thanks! Fixed up